refactor: переделал авторизацию для доступа к Google Spreadsheet

VladBaykin opened the pull request:
Авторизация к Google Spreadsheet через статичный credentials.json
- изменен порт в application-local-stack.yml на 8080
- в README добавлена инструкция по credentials

VladBaykin committed:
refactor: переделал авторизацию для доступа к Google Spreadsheet

VladBaykin committed:
fix: поправил docker-build.yml

VladBaykin committed:
fix: изменил порт в application-local-stack.yml

VladBaykin committed:
chore: добавил в README.md инструкцию по добавлению credentials.json

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/dataimporter/google/GoogleSheetsClient.java:
```diff
@@ -1,47 +1,82 @@
+            }
+        }
+
+        String credentialsJson = System.getenv("GOOGLE_APPLICATION_CREDENTIALS_JSON");
```
Лучше сделать более явно через `@Value("${}")`

DarkRubin commented on src/test/java/com/itmentorcommunityplatform/dataimporter/DataImporterApplicationTests.java:
```diff
@@ -1,13 +1,11 @@
-
-}
+//@SpringBootTest
+//class DataImporterApplicationTests {
```
Можно было бы придумать какие-то тестовые значения конфигурации и/или замокать бин гугл клиента. А вообще нужно было просто удалить тогда этот класс тестов если он не нужен, а не оставлять закомментированным

DarkRubin commented on README.md:
````diff
@@ -17,43 +17,17 @@
+2. Запуск
    - Через консоль
    ```bash
    ./gradlew bootRun --args='--spring.profiles.active=ide'
````
Если указывать профиль в аргументах, то в .env можно не дублировать

DarkRubin commented on src/main/java/com/itmentorcommunityplatform/dataimporter/google/GoogleSheetsClient.java:
```diff
@@ -1,47 +1,82 @@
+    public void init() {
+        try {
+            GoogleCredentials credentials = loadCredentials();
+            credentials = credentials.createScoped(Collections.singletonList(SCOPE_SHEETS_READONLY));
```
Можно использовать просто значение из enum'a `SheetsScopes.DRIVE_READONLY`

DarkRubin reviewed (changes requested):
По функционалу все ок, поправь мелкие моменты

VladBaykin committed:
refactor: корректная загрузка Google credentials через @Value, обновление README и .env
- заменена загрузка credentials через System.getenv на @Value
- использован SheetsScopes.SPREADSHEETS_READONLY вместо строковой константы
- обновлены README и .env
